presto-parquet/src/test/java/com/facebook/presto/parquet/writer/TestParquetWriter.java

```diff
@@ -0,0 +1,111 @@
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.List;
+import java.util.Random;
+import java.util.UUID;
+
+import static com.facebook.presto.common.type.BigintType.BIGINT;
+import static com.facebook.presto.common.type.BooleanType.BOOLEAN;
+import static com.facebook.presto.common.type.IntegerType.INTEGER;
+import static com.facebook.presto.common.type.VarcharType.VARCHAR;
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.io.Files.createTempDir;
+import static com.google.common.io.MoreFiles.deleteRecursively;
+import static com.google.common.io.RecursiveDeleteOption.ALLOW_INSECURE;
+import static java.util.UUID.randomUUID;
+import static org.testng.Assert.fail;
+
+public class TestParquetWriter
+{
+    private File temporaryDirectory;
+    private File parquetFile;
+
+    @BeforeMethod
+    public void setup()
+    {
+        temporaryDirectory = createTempDir();
+        parquetFile = new File(temporaryDirectory, randomUUID().toString());
+    }
+
+    @AfterMethod
+    public void tearDown()
+            throws IOException
+    {
+        deleteRecursively(temporaryDirectory.toPath(), ALLOW_INSECURE);
+    }
+
+    @Test
+    public void testRowGroupFlushInterleavedColumnWriterFallbacks()
+    {
+        List<Type> types = ImmutableList.of(BIGINT, INTEGER, VARCHAR, BOOLEAN);
+        List<String> names = ImmutableList.of("col_1", "col_2", "col_3", "col_4");
+        ParquetWriterOptions parquetWriterOptions = ParquetWriterOptions.builder()
+                .setMaxPageSize(DataSize.succinctBytes(1000))
+                .setMaxBlockSize(DataSize.succinctBytes(15000))
+                .setMaxDictionaryPageSize(DataSize.succinctBytes(1000))
+                .build();
+        try (ParquetWriter parquetWriter = createParquetWriter(parquetFile, types, names, parquetWriterOptions, CompressionCodecName.UNCOMPRESSED)) {
+            Random rand = new Random();
+            for (int pageIdx = 0; pageIdx < 10; pageIdx++) {
+                int pageRowCount = 100;
+                PageBuilder pageBuilder = new PageBuilder(pageRowCount, types);
+                for (int rowIdx = 0; rowIdx < pageRowCount; rowIdx++) {
+                    // maintain col_1's dictionary size approximately half of raw data
+                    BIGINT.writeLong(pageBuilder.getBlockBuilder(0), pageIdx * 100 + rand.nextInt(50));
+                    INTEGER.writeLong(pageBuilder.getBlockBuilder(1), rand.nextInt(100000000));
+                    VARCHAR.writeString(pageBuilder.getBlockBuilder(2), UUID.randomUUID().toString());
+                    BOOLEAN.writeBoolean(pageBuilder.getBlockBuilder(3), rand.nextBoolean());
+                    pageBuilder.declarePosition();
+                }
+                parquetWriter.write(pageBuilder.build());
+            }
+        }
+        catch (Exception e) {
+            fail("Should not fail!");
```

Comment:
The error message doesn't add much value for troubleshooting.

Reply:
Fixed.